cmake/modules/compiler_config.cmake

```diff
@@ -102,6 +102,12 @@ if (C_SUPPORTS_NO_FORMAT_TRUNCATION)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-format-truncation")
 endif()
 
+check_c_compiler_flag("-Wno-unused-parameter"
+  C_SUPPORTS_NO_FORMAT_TRUNCATION)
+if (C_SUPPORTS_NO_FORMAT_TRUNCATION)
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-parameter")
+endif()
+
```

Comment:
I believe the CMake variable should be named something like `C_SUPPORTS_NO_UNUSED_PARAMETER` instead of `C_SUPPORTS_NO_FORMAT_TRUNCATION`?